spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb

```diff
@@ -14,92 +14,129 @@
 
     credentials = {:userid => username, :password => password}
     @ems.authentications << FactoryBot.create(:authentication, credentials)
+    @inventory = nil
+  end
+
+  def reset_cache
+    @ems.reset_openstack_handle
+
+    require "fog/openstack"
+    Fog::OpenStack.instance_variable_set(:@version, nil)
+  end
+
+  def refresh_inventory
+    with_vcr do
+      reset_cache
+
+      @ems.refresh
+      @ems.network_manager.refresh
+      @ems.cinder_manager.refresh
+      @ems.swift_manager.refresh
+    end
+
+    @inventory ? assert_inventory_not_changed { @inventory } : @inventory = serialize_inventory
+    @ems.reload
+
+    assert_specific_vm
+    assert_specific_auth_key_pair
+    assert_specific_cloud_volume
+    assert_specific_host_aggregates
+    assert_specific_cloud_network
+    assert_specific_cloud_subnet
+    assert_specific_security_group
+    assert_specific_network_router
```

Comment:
I wonder if we should split up the "refresh_inventory" from the checks since different tests might want to check different things.  We can still move all of these into an `assert_common` to reduce the duplication like you've done here

Reply:
> we should split up the "refresh_inventory" from the checks **since different tests might want to check different things**

Wouldn't moving all the checks into an `assert_common` prevent this? Since all the checks are in one place

Reply:
Also not sure how we would go about asserting certain checks while leaving out others depending on the test